### ISSUE TITLE
How to use multi thread to display multiple cameras on windows ?

### ISSUE THREAD

**Issue by WestbrookZero** [CMU-Perceptual-Computing-Lab/openpose#1877]
Using either method or the second method will not work, the second camera will not show.
the first method as follow:
`for (uint8_t i = 0; i < useCameraAmount; i++)
	{
		std::thread t(openPoseDemo, m_CameraConfigs[i]);
		t.detach();
	}
`
	
the second method as follow:
        op::ProducerType producerType01;
	op::String producerString01;
	op::ProducerType producerType02;
	op::String producerString02;
	std::tie(producerType01, producerString01) = op::flagsToProducer(
	op::String(FLAGS_image_dir), op::String(FLAGS_video), op::String(FLAGS_ip_camera), 0,
	FLAGS_flir_camera, FLAGS_flir_camera_index);
	const auto displayProducerFpsMode01 = (FLAGS_process_real_time
	? op::ProducerFpsMode::OriginalFps : op::ProducerFpsMode::RetrievalFps);
	auto producerSharedPtr01 = createProducer(
	producerType01, producerString01.getStdString(), cameraSize, FLAGS_camera_parameter_path, FLAGS_frame_undistort,
	FLAGS_3d_views);
	producerSharedPtr01->setProducerFpsMode(displayProducerFpsMode01);
	std::tie(producerType02, producerString02) = op::flagsToProducer(
	op::String(FLAGS_image_dir), op::String(FLAGS_video), op::String(FLAGS_ip_camera), 1,
	FLAGS_flir_camera, FLAGS_flir_camera_index);
	const auto displayProducerFpsMode02 = (FLAGS_process_real_time
	? op::ProducerFpsMode::OriginalFps : op::ProducerFpsMode::RetrievalFps);
	auto producerSharedPtr02 = createProducer(
	producerType02, producerString02.getStdString(), cameraSize, FLAGS_camera_parameter_path, FLAGS_frame_undistort,
	FLAGS_3d_views);
	producerSharedPtr02->setProducerFpsMode(displayProducerFpsMode02);
	auto DatumProducer01 = std::make_shared<op::DatumProducerop::Datum>(producerSharedPtr01);
	auto wDatumProducer01 = std::make_shared<op::WDatumProducerop::Datum>(DatumProducer01);

	auto DatumProducer02 = std::make_shared<op::DatumProducer<op::Datum>>(producerSharedPtr02);
	auto wDatumProducer02 = std::make_shared<op::WDatumProducer<op::Datum>>(DatumProducer02);

	// Specific WUserClass
	auto wUserClass = std::make_shared<WUserClass>();

	// GUI (Display)
	auto gui01 = std::make_shared<op::Gui>(outputSize, FLAGS_fullscreen, threadManager.getIsRunningSharedPtr());
	/*auto gui01 = std::make_shared<op::Gui>(outputSize, FLAGS_fullscreen, producerSharedPtr01);*/
	auto wGui01 = std::make_shared<op::WGui<TypedefDatumsSP>>(gui01);

	auto gui02 = std::make_shared<op::Gui>(outputSize, FLAGS_fullscreen, threadManager.getIsRunningSharedPtr());
	/*auto gui02 = std::make_shared<op::Gui>(outputSize, FLAGS_fullscreen, producerSharedPtr02);*/
	auto wGui02 = std::make_shared<op::WGui<TypedefDatumsSP>>(gui02);

	auto threadId00 = 0ull;
	auto queueIn00 = 0ull;
	auto queueOut00 = 1ull;
	auto threadId01 = 0ull;
	auto queueIn01 = 0ull;
	auto queueOut01 = 1ull;
	auto threadId02 = 0ull;
	auto queueIn02 = 0ull;
	auto queueOut02 = 1ull;
	threadManager.add(threadId00++, wDatumProducer01, queueIn01++, queueOut01++);
	threadManager.add(threadId00++, wGui01, queueIn01++, queueOut01++);
	threadManager.add(threadId00++, wDatumProducer02, queueIn01++, queueOut01++);
	threadManager.add(threadId00++, wGui01, queueIn01++, queueOut01++);
